pandas/core/frame.py

```diff
@@ -3211,6 +3211,14 @@ def _setitem_slice(self, key: slice, value):
         self._check_setitem_copy()
         self.iloc[key] = value
 
+    def _check_key_length(self, key, value: DataFrame):
```

Comment:
maybe put in core.indexers and pass self.columns?

can this be reused anywhere else? (IIRC you recently implemented another get_indexer_non_unique check in a similar context)

Reply:
+1 for this. ideally we move more and more indexing code out of here

Reply:
Moved.

Unfortunately they only have the get_indexer_for call in common. The equality check is quite different